ompi/mpi/c/comm_set_info.c

```diff
@@ -46,7 +46,13 @@ int MPI_Comm_set_info(MPI_Comm comm, MPI_Info info)
         }
     }
 
+    /* protect communication posted earlier from inconsistent info states */
+    comm->c_coll->coll_barrier (comm, comm->c_coll->coll_barrier_module );
```

Comment:
I have concern about this change.  I agree that the standard allows for it, but I'm not aware of any MPI implementation that actually enforces the collective nature of MPI_Comm_set_info.  There's a FIXME note in MPICH that they should, to ensure consistency, but no actual enforcement.  We have no idea what codes we'd break by adding this code.  Given how long we've shipped the allow_overtaking flag, I'm not sure I have a great solution to this problem.

If we are going to risk breaking codes, shouldn't we also switch from a barrier to some form of consistency checking check?